testnet/stacks-node/src/neon_node.rs

```diff
@@ -283,17 +308,60 @@ fn spawn_miner_relayer(mut relayer: Relayer, local_peer: LocalPeer,
     let burn_fee_cap = config.burnchain.burn_fee_cap;
     let mut bitcoin_controller = BitcoinRegtestController::new_dummy(config);
 
-
     let blocks_path = chainstate.blocks_path.clone();
+    let mut block_on_recv = false;
 
     let _relayer_handle = thread::spawn(move || {
-        while let Ok(mut directive) = relay_channel.recv() {
+        loop {
```

Comment:
Can this be changed back to a while loop? If you add a `RelayerDirective::TryProcessAttachable` variant (note the spelling 😉), this could be written as:

```rust
while let Ok(mut directive) =
    if block_on_recv {
       relay_channel.recv()
    } else { 
       relay_channel.try_recv().or_else( |e|
          match e {
              TryRecvError::Empty => Ok(RelayerDirective::TryProcessAttachable),
              other => Err(other)
          })
      } {
   block_on_recv = false;
   match directive {
      RelayerDirective::TryProcessAttachable => {
          // process any attachable blocks
...
```

It's a little complex of a loop condition, but I think it makes what's happening in the loop more clear (all the work is getting done in the match, and we wouldn't need breaks or continues).

Reply:
>  (note the spelling )
Wee cant splel ovre hear.  Will ficks teh spleling.

In the man time, I can add a `TryProcessAttachable` variant.